### ISSUE TITLE
[Bug]: Mail app Office365 authentification not working

### ISSUE THREAD

**Issue by Ranily57** [nextcloud/mail#9139]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I tried a lot of thing i found on the internet, but i still can't put my office365 mail into the Mail App.

### Steps to reproduce

1. Login to your Nextcloud
2. Go to Mail App
3. Authentificate with the automatic login
4. it's not working but it put all the nessesary information into the manual part
5. Try the manual connexion

### Expected behavior

I expect the app to work, get my mail into the mail app and work with it.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
<?php
$CONFIG = array (
  'instanceid' => '****',
  'passwordsalt' => '****',
  'secret' => '***',
  'trusted_domains' =>
  array (
    0 => '*****',
  ),
  'datadirectory' => '/var/www/nextcloud/data',
  'dbtype' => 'mysql',
  'version' => '27.1.4.1',
  'overwrite.cli.url' => 'http://nc.virtualshowproduction.com:81',
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,
  'dbuser' => 'nextcloud',
  'dbpassword' => '*****',
  'installed' => true,
  'twofactor_enforced' => 'false',
  'twofactor_enforced_groups' =>
  array (
    0 => 'VSP',
  ),
  'twofactor_enforced_excluded_groups' =>
  array (
  ),
  'mail_from_address' => '**.**',
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_domain' => '***',
  'mail_smtphost' => 'smtp.office365.com',
  'mail_smtpport' => '587',
  'maintenance' => false,
  'debug' => true,
  'app.mail.smtp.timeout' => 60,
  'app.mail.transport' => 'php-mail',
);
```

### List of activated Apps

```shell
Command not working
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by ChristophWurst** [comment]
> ### Nextcloud Logs
> 
> _No response_

**Comment by Ranily57** [comment]
Oh i forgot them, there is the last log form nextcloud.log: 

``` {"reqId":"tx0Zj2vk5UktO8ya0qfR","level":3,"time":"2023-12-05T05:17:24+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"tx0Zj2vk5UktO8ya0qfR","level":3,"time":"2023-12-05T05:17:30+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"tx0Zj2vk5UktO8ya0qfR","level":3,"time":"2023-12-05T05:17:30+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"tx0Zj2vk5UktO8ya0qfR","level":3,"time":"2023-12-05T05:17:35+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"tx0Zj2vk5UktO8ya0qfR","level":3,"time":"2023-12-05T05:17:42+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:17+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"Undefined array key 1 at /var/www/nextcloud/lib/private/AppFramework/Utility/ControllerMethodReflector.php#65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:17+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"Undefined array key 1 at /var/www/nextcloud/lib/private/AppFramework/Utility/ControllerMethodReflector.php#65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:18+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:18+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:24+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:24+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:29+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"uQEekycTsF7xpV7XsLet","level":3,"time":"2023-12-05T05:27:36+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:29+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"Undefined array key 1 at /var/www/nextcloud/lib/private/AppFramework/Utility/ControllerMethodReflector.php#65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:29+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"Undefined array key 1 at /var/www/nextcloud/lib/private/AppFramework/Utility/ControllerMethodReflector.php#65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:30+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:30+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:36+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:36+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:41+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"R6JV1rRpLTTFGME5lIFl","level":3,"time":"2023-12-05T06:00:48+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/autoconfig/ispdb/*******.**********%40********.**","message":"dns_get_record(): A temporary server error occurred. at /var/www/nextcloud/lib/private/Http/Client/DnsPinMiddleware.php#84","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"JDEkXMDiRGLRfcHD2zhd","level":3,"time":"2023-12-05T06:08:00+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/settings/admin/serverinfo","message":"file_get_contents(): Read of 8192 bytes failed with errno=22 Invalid argument at /var/www/nextcloud/apps/serverinfo/lib/OperatingSystems/DefaultOs.php#254","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"JDEkXMDiRGLRfcHD2zhd","level":3,"time":"2023-12-05T06:08:00+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"PHP","method":"GET","url":"/index.php/settings/admin/serverinfo","message":"file_get_contents(): Read of 8192 bytes failed with errno=22 Invalid argument at /var/www/nextcloud/apps/serverinfo/lib/OperatingSystems/DefaultOs.php#254","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"PHP"}}
{"reqId":"krz9rnu71I8aKdOEndKx","level":2,"time":"2023-12-05T06:08:01+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"settings","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Controller OCA\\Settings\\Controller\\CheckSetupController::check executed 4740 queries.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"settings"}}
{"reqId":"JhRTefmNpKUOIBRI5Ooq","level":2,"time":"2023-12-05T06:08:04+00:00","remoteAddr":"**.**.**.**","user":"********* *******","app":"settings","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Controller OCA\\Settings\\Controller\\CheckSetupController::check executed 4740 queries.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"27.1.4.1","data":{"app":"settings"}}```

**Comment by Ranily57** [comment]
Fixing my DNS ? how i do that ?

**Comment by ChristophWurst** [comment]
Try https://help.nextcloud.com/c/support/7. I can not offer you support for this.

**Comment by Ranily57** [comment]
J'ai ajouter les adresses dans le `resolv.conf` mais ca change toujours rien. 

J'ai ajouté `1.1.1.1` et `1.0.0.1` et `127.0.0.1`

**Comment by Ranily57** [comment]
So when i try to log on the mail app, i get `The username or the IMAP password is incorrect`, it is not, because i use my Google Account to put the password on it, i'm sure it's not true. But actually i got those other logs:

![image](https://github.com/nextcloud/mail/assets/62222362/5d65a9b7-edf0-4fb0-9819-daa085f4b753)

**Comment by joshtrichards** [comment]
@Ranily57 https://github.com/nextcloud/mail/issues/9139#issuecomment-1840274373

Please ask for some assistance on the Nextcloud Help Forum. If nothing else they'll be able to help you confirm DNS is truly working so that we're not troubleshooting your environment while also troubleshooting a potential bug.

Those errors are coming from Nextcloud Server not the Mail app directly. So chances are if there even is a bug, it's not in Mail. But I suspect it's a local environment/config matter. See you over at the Help Forum! :-)

**Comment by joshtrichards** [comment]
The "undefined array key 1 [...] in ControllerMethodReflector.php" is fixed is #9216 / #9226

**Comment by Ranily57** [comment]
The problem is solved by using the Microsoft OAuth, thank's for your answer !
